### PR TITLE
PWX-26914: Add resources config for stork, stork-scheduler and autopilot

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Pure-AdamuKaapan — COMMENTED** [review]
Makes sense to me, will let someone else review as well to sanity check

File: deploy/crds/core_v1_storagecluster_crd.yaml [hunk 1]
```diff
@@ -643,6 +643,30 @@ spec:
                                               type: string
                                             mode:
                                               type: integer
+                  resources:
+                    type: object
+                    description: Specifies the compute resource requirements for the storage pod.
```
Comment (pure-jliao) [inline deploy/crds/core_v1_storagecluster_crd.yaml:648]: guess you just copied it from storage pod resource setting? need to rephrase those descriptions.

File: pkg/controller/storagecluster/stork.go [hunk 2]
```diff
@@ -1117,6 +1121,12 @@ func getStorkSchedDeploymentSpec(
 		deployment.Spec.Template.Spec.TopologySpreadConstraints = topologySpreadConstraints
 	}
 
+	// If resources is specified in the spec, the resources specified by annotation (such as portworx.io/stork-cpu)
```
Comment (pure-jliao) [inline pkg/controller/storagecluster/stork.go:1124]: there are 2 annotations stork-cpu and stork-scheduler-cpu, they might be different, after the spec change we unify them into one?

**Review by pure-jliao — APPROVED** [review]
LGTM. have one suggestion how we can handle the old annotations. I'm thinking we can remove the annotation check logic in autopilot and stork when setting the deployment specs, then check the old annotations in "set portworx default" function, if exists we set the new resource spec values, and remove the annotation from stc, so we can safely deprecate the annotations. only thing is stork has 2 separate annotations we may need to combine them into 1.

File: pkg/controller/storagecluster/stork.go (hunk 2 above)
Comment (haibinxie) [reply pkg/controller/storagecluster/stork.go:1124]: Yes, in storagecluster spec we can't configure them separately

File: deploy/crds/core_v1_storagecluster_crd.yaml (hunk 1 above)
Comment (haibinxie) [reply deploy/crds/core_v1_storagecluster_crd.yaml:648]: nice catch.

**Comment by haibinxie** [comment]
> LGTM. have one suggestion how we can handle the old annotations. I'm thinking we can remove the annotation check logic in autopilot and stork when setting the deployment specs, then check the old annotations in "set portworx default" function, if exists we set the new resource spec values, and remove the annotation from stc, so we can safely deprecate the annotations. only thing is stork has 2 separate annotations we may need to combine them into 1.

Changing STC may cause confusion to the users.

**Pull request merged**